crates/shared/src/arguments.rs

```diff
@@ -260,6 +260,13 @@ pub struct Arguments {
     /// Override address of the balancer vault contract.
     #[clap(long, env)]
     pub balancer_v2_vault_address: Option<H160>,
+
+    /// EOA account used to sign transactions on behalf of the authenticator
+    /// manager.
+    ///
+    /// Manual transaction adding the role for this EOA is assumed to be done
+    #[clap(long, env)]
+    pub authenticator_eoa: H256,
```

Comment:
If we only plan to run the circuit breaker in the autopilot the arguments should not be in the shared crate. Also I assume there will be at least a few more arguments related to this so it probably makes sense to factor this into a separate struct that gets flattened into the main autopilot arguments struct.

Reply:
On second thought without the eoa argument stuff this is already a fairly self contained PR. Could make sense to only add it onces it's actually needed to avoid putting it in the wrong location. (Putting the `Roles` into the `Contracts` struct seems like a safe choice)

Reply:
@MartinquaXD I removed the configuration, make it optional (so it is opt-in functionality as @fleupold suggested), and set this PR to `None`, so we can merge @sunce86 's work and I can continue from here by implementing the configuration (adding also solver's addresses as per https://github.com/cowprotocol/services/issues/2667#issuecomment-2104417780) and tests. This way this PR is a standalone PR as you suggested. WDYT?